packages/@aws-cdk/aws-stepfunctions/README.md

```diff
@@ -1,2 +1,322 @@
+
+This library comes with a set of classes that model the [Amazon States
+Language](https://states-language.net/spec.html). The following State classes
+are supported:
+
+* `Task`
+* `Pass`
+* `Wait`
+* `Choice`
+* `Parallel`
+* `Succeed`
+* `Fail`
+
+An arbitrary JSON object (specified at execution start) is passed from state to
+state and transformed during the execution of the workflow. For more
+information, see the States Language spec.
+
+### Task
+
+A `Task` represents some work that needs to be done. It takes a `resource`
+property that is either a Lambda `Function` or a Step Functions `Activity`.
+
+```ts
+const task = new stepfunctions.Task(this, 'Invoke The Lambda', {
+    resource: myLambda,
+    inputPath: '$.input',
+    timeoutSeconds: 300,
+});
+
+// Add a retry policy
+task.retry({
+    intervalSeconds: 5,
+    maxAttempts: 10
+});
+
+// Add an error handler
+task.onError(errorHandlerState);
+
+// Set the next state
+task.next(nextState);
+```
+
+### Pass
+
+A `Pass` state does no work, but it can optionally transform the execution's
+JSON state.
+
+```ts
+// Makes the current JSON state { ..., "subObject": { "hello": "world" } }
+const pass = new stepfunctions.Pass(this, 'Add Hello World', {
+    result: { hello: "world" },
+    resultPath: '$.subObject',
+});
+
+// Set the next state
+pass.next(nextState);
+```
+
+### Wait
+
+A `Wait` state waits for a given number of seconds, or until the current time
+hits a particular time. The time to wait may be taken from the execution's JSON
+state.
+
+```ts
+// Wait until it's the time mentioned in the the state object's "triggerTime"
+// field.
+const wait = new stepfunctions.Wait(this, 'Wait For Trigger Time', {
+    timestampPath: '$.triggerTime',
+});
+
+// Set the next state
+wait.next(startTheWork);
+```
+
+### Choice
+
+A `Choice` state can take a differen path through the workflow based on the
+values in the execution's JSON state:
+
+```ts
+const choice = new stepfunctions.Choice(this, 'Did it work?');
+
+// Add conditions with .on()
+choice.on(stepfunctions.Condition.stringEqual('$.status', 'SUCCESS'), successState);
+choice.on(stepfunctions.Condition.numberGreaterThan('$.attempts', 5), failureState);
+
+// Use .otherwise() to indicate what should be done if none of the conditions match
+choice.otherwise(tryAgainState);
+```
+
+If you want to temporarily branch your workflow based on a condition, but have
+all branches come together and continuing as one (similar to how an `if ...
+then ... else` works in a programming language), use the `.afterwards()` method:
+
+```ts
+const choice = new stepfunctions.Choice(this, 'What color is it?');
+choice.on(stepfunctions.Condition.stringEqual('$.color', 'BLUE'), handleBlueItem);
+choice.on(stepfunctions.Condition.stringEqual('$.color', 'RED'), handleRedItem);
+choice.otherwise(handleOtherItemColor);
+
+// Use .afterwards() to join all possible paths back together and continue
+choice.afterwards().next(shipTheItem);
```

Comment:
In the upstream language, `Choice` doesn't have this "joining all possible branches back together" feature, and I didn't want to force this on people in case they build a state machine they didn't intend to. Also, `afterwards()` takes options (not pictured here) that allow you to exert some control on how you want to pick selectable paths. For example, you have to pick whether or not (default: not) to include error transitions.